x-pack/plugins/apm/public/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState, useEffect } from 'react';
+
+export function useLocalStorage<T>(key: string, defaultValue: T) {
+  const [item, setItem] = useState<T>(getFromStorage());
+
+  function getFromStorage() {
+    const storedItem = window.localStorage.getItem(key);
+
+    let toStore: T = defaultValue;
+
+    if (storedItem !== null) {
+      try {
+        toStore = JSON.parse(storedItem) as T;
+      } catch (err) {
+        // do nothing
```

Comment:
In what cases does it make sense to swallow the error?

Reply:
Users can do whatever they want with localStorage, in that case it's pretty punishing to break the UI. I could also log it but it can be noisy, as this is a hook that is called every render. What do you suggest as appropriate here?

Reply:
What about deleting the invalid locale storage item and logging it. This way we'll recover and tell the user that something went wrong.

Reply:
```suggestion
        window.localStorage.removeItem(key);
        console.log(`Unable to decode: ${storedItem}`)
```

Reply:
My bad, missed this - I've implemented your suggestion (incoming).